AdobeStockImageAdminUi/Controller/Adminhtml/System/Config/TestConnection.php

```diff
@@ -0,0 +1,119 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\AdobeStockImageAdminUi\Controller\Adminhtml\System\Config;
+
+use Exception;
+use Magento\AdobeStockImageApi\Api\GetImageListInterface;
+use Magento\Backend\App\Action;
+use Magento\Backend\App\Action\Context;
+use Magento\Framework\Api\Search\SearchCriteriaBuilder;
+use Magento\Framework\App\ResponseInterface;
+use Magento\Framework\Controller\Result\Json;
+use Magento\Framework\Controller\Result\JsonFactory;
+use Magento\Framework\Controller\ResultInterface;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Exception\NotFoundException;
+use Magento\Framework\Filter\StripTags;
+
+class TestConnection extends Action
+{
+    /**
+     * Authorization level of a basic admin session.
+     *
+     * @see _isAllowed()
+     */
+    const ADMIN_RESOURCE = 'Magento_AdobeStockImageAdminUi::adobe_stock_integration';
+
+    /**
+     * @var JsonFactory
+     */
+    private $resultJsonFactory;
+
+    /**
+     * @var StripTags
+     */
+    private $tagFilter;
+
+    /**
+     * @var GetImageListInterface
+     */
+    private $getImageList;
+
+    /**
+     * @var SearchCriteriaBuilder
+     */
+    private $searchCriteriaBuilder;
+
+    /**
+     * TestConnection constructor.
+     *
+     * @param Context               $context
+     * @param GetImageListInterface $getImageList
+     * @param SearchCriteriaBuilder $searchCriteriaBuilder
+     * @param JsonFactory           $resultJsonFactory
+     * @param StripTags             $tagFilter
+     */
+    public function __construct(
+        Context $context,
+        GetImageListInterface $getImageList,
+        SearchCriteriaBuilder $searchCriteriaBuilder,
+        JsonFactory $resultJsonFactory,
+        StripTags $tagFilter
+    ) {
+        parent::__construct($context);
+        $this->getImageList = $getImageList;
+        $this->searchCriteriaBuilder = $searchCriteriaBuilder;
+        $this->resultJsonFactory = $resultJsonFactory;
+        $this->tagFilter = $tagFilter;
+    }
+
+    /**
+     * Execute action based on request and return result
+     *
+     * @return ResultInterface|ResponseInterface
+     * @throws NotFoundException
+     */
+    /**
+     * Check for connection to server
+     *
+     * @return Json
+     */
+    public function execute()
+    {
+        $result = [
+            'success'      => false,
+            'errorMessage' => '',
+        ];
+
+        try {
+            $response = $this->getImageList->execute($this->getSearchCriteria());
```

Comment:
Considering that the configuration is general and not specifically related to image I would avoid dependency on AdobeStockImage module. For this purpose, I propose to introduce a `getAssetList` or `testConnection` API to AdobeStockAssetApi and use the call to that webapi endpoint

Reply:
Hey guys, I have applied the requested changes. Could you please review when you get a chance?

**Note:** I am not very happy with the implementation of `testConnection` method, can you see a better way to implement it?

Reply:
Thanks for the contribution, @diazwatson ! I think test connection implementation is perfect for now, also, we can improve it later if there will be ideas

Reply:
Thanks @sivaschenko I'll change that tonight

Reply:
@diazwatson actually, I did this already. Pull request is merged! Please take a look at my changes and feel free to create a new pull request if you'd like to adjust anything